RecoHGCal/TICL/plugins/TracksterLinkingbySkeletons.cc

```diff
@@ -0,0 +1,417 @@
+    LogDebug("TracksterLinkingbySkeletons") << "Inner Trackster with energy " << myTrackster.raw_energy() << " Num LCs "
+                                            << myTrackster.vertices().size() << " NOT GOOD " << std::endl;
+    return false;
+  } else {
+    LogDebug("TracksterLinkingbySkeletons") << "Inner Trackster wi energy " << myTrackster.raw_energy() << " Num LCs "
+                                            << myTrackster.vertices().size() << " IS GOOD " << std::endl;
+    float proj_distance = projective_distance(mySkeleton[1], otherSkeleton[1]);
+    auto isEE = mySkeleton[1].z() <= zVal_interface ? 0 : 1;
+    bool areAlignedInProjectiveSpace = proj_distance < max_distance_projective_sqr_[isEE];
+    LogDebug("TracksterLinkingbySkeletons")
+        << "\t Trying to compare with outer Trackster with energy " << otherTrackster.raw_energy() << " Num LCS "
+        << otherTrackster.vertices().size() << " Projective distance " << proj_distance << " areAlignedProjective "
+        << areAlignedInProjectiveSpace << " TH " << max_distance_projective_sqr_[isEE] << std::endl;
+    //check if otherTrackster is good
+    if (isGoodTrackster(otherTrackster, otherSkeleton, min_num_lcs_, min_trackster_energy_, pca_quality_th_)) {
+      // if both tracksters are good, then we can check the projective distance between the barycenters.
+      // if the barycenters are aligned, then we check that the two skeletons are aligned
+      if (areAlignedInProjectiveSpace) {
+        auto dotProdSkeletons =
+            ((mySkeleton[2] - mySkeleton[0]).Unit()).Dot((otherSkeleton[2] - otherSkeleton[0]).Unit());
+        bool alignedSkeletons = dotProdSkeletons > dot_prod_th_;
+        LogDebug("TracksterLinkingbySkeletons")
+            << "\t Outer Trackster is Good, checking for skeleton alignment " << alignedSkeletons << " dotProd "
+            << dotProdSkeletons << " Threshold " << dot_prod_th_ << std::endl;
+        if (alignedSkeletons)
+          LogDebug("TracksterLinkingbySkeletons") << "\t\t Linked! " << std::endl;
+        return alignedSkeletons;
+      } else {
+        // we measure the distance between the two closest nodes in the two skeletons
+        LogDebug("TracksterLinkingbySkeletons")
+            << "\t Outer Trackster is not aligned,  check skeletons distances " << std::endl;
+        int myClosestPoint = -1;
+        int otherClosestPoint = -1;
+        float minDistance_z = std::numeric_limits<float>::max();
+        for (int i = 0; i < 3; i++) {
+          for (int j = 0; j < 3; j++) {
+            float dist_z = std::abs(mySkeleton[i].Z() - otherSkeleton[j].Z());
+            if (dist_z < minDistance_z) {
+              myClosestPoint = i;
+              otherClosestPoint = j;
+              minDistance_z = dist_z;
+            }
+          }
+        }
+        LogDebug("TracksterLinkingbySkeletons")
+            << "\t\t Check distance in Z " << minDistance_z << " TH " << min_distance_z_ << std::endl;
```

Comment:
src/RecoHGCal/TICL/plugins/TracksterLinkingbySkeletons.cc:235:71: note: in instantiation of function template specialization 'edm::Log<edm::level::Debug, false>::operator<<<std::vector<double>>' requested here
  235 |             << "\t\t Check distance in Z " << minDistance_z << " TH " << min_distance_z_ << std::endl;

Reply:
min_distance_z_ is vector of double, right?

Reply:
thanks, Phat. I removed that line, it was used only when developing and was left in not working conditions.